### PR TITLE
Change default font in desktop settings to system fonts

### PR DESCRIPTION
**Pull request by aguscruiz**
Removing Poppins as the font for the settings pages in favor of using the users' default system font. This will make the pages feel a bit more native and modern since we're moving away from Poppins for texts

Resolves https://github.com/brave/brave-browser/issues/41559

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

1. Font should match the System font on the chrome://settings page

### CODE REVIEW

File: ui/webui/resources/css/text_defaults_md.css [hunk 1]
```diff
@@ -19,7 +19,7 @@ html {
 }
 
 body {
-  font-family: Poppins, $i18nRaw{fontfamily};
+  font-family: $i18nRaw{fontfamily};
```
Comment (fallaciousreasoning) [inline ui/webui/resources/css/text_defaults_md.css:22]: Do we need to specify `system-ui` here, or is that the default?

Comment (petemill) [reply ui/webui/resources/css/text_defaults_md.css:22]:
```
fontfamily: "system-ui, sans-serif",
fontfamilyMd: "system-ui, sans-serif"
```

Comment (aguscruiz) [reply ui/webui/resources/css/text_defaults_md.css:22]: I **think** that's the default when you don't specify it

File: browser/resources/settings/brave_overrides/settings_section.ts [hunk 2]
```diff
@@ -10,6 +10,9 @@ RegisterStyleOverride(
   'settings-section',
   html`
     <style include="settings-shared">
+    #header .title {
+    font-weight: 500 !important;
+    }
```
Comment (fallaciousreasoning) [inline browser/resources/settings/brave_overrides/settings_section.ts:15]:
```suggestion
      #header .title {
        font-weight: 500 !important;
      }
```

**Comment by fallaciousreasoning** [comment]
Okay @aguscruiz could you create an issue for this (and maybe lets not add more to this PR) :smile:

File: browser/resources/settings/brave_overrides/settings_menu.ts [hunk 3]
```diff
@@ -63,7 +63,8 @@ RegisterStyleOverride(
         border-radius: 6px;
         background-color: #fff;
         overflow-y: auto;
-        padding: 30px !important;
+        padding: 24px !important;
+        box-shadow: 0px var(--Elevation-xxs, 1px) 0px 0px var(--Semantic-Elevation-Primary, rgba(0, 0, 0, 0.05)), 0px var(--Elevation-xxs, 1px) var(--Elevation-xs, 4px) 0px var(--Semantic-Elevation-Secondary, rgba(0, 0, 0, 0.10));
```
Comment (fallaciousreasoning) [inline browser/resources/settings/brave_overrides/settings_menu.ts:67]: nit: 
```suggestion
        box-shadow: var(--leo-effect-elevation-01);
```

**Review by fallaciousreasoning — APPROVED** [review]
lgtm with 1 nit

**Comment by aguscruiz** [comment]
Resolves https://github.com/brave/brave-browser/issues/41559

**Comment by aguscruiz** [comment]
resolves https://github.com/brave/brave-browser/issues/41559

**Pull request merged**